relevanceai/api/batch/batch_insert.py

```diff
@@ -755,6 +757,309 @@ def terminate(self):
+
+        Parameters
+        ----------
+        dataset_id: str
+            The dataset_id of the collection where your original documents are
+
+        update_function
+            A function created by you that converts documents in your original collection into the updated documents. The function must contain a field which takes in a list of documents from the original collection. The output of the function must be a list of updated documents.
+
+        updating_args: dict
+            Additional arguments to your update_function, if they exist. They must be in the format of {'Argument': Value}
+
+        updated_dataset_id: str
+            The dataset_id of the collection where your updated documents are uploaded into. If 'None', then your original collection will be updated.
+
+        log_file: str
+            The log file to direct any information or issues that may crop up.
+            If no log file is specified, one will automatically be created.
+
+        retrieve_chunk_size: int
+            The number of documents that are received from the original collection with each loop iteration.
+
+        filters: list
+            A list of filters to apply on the retrieval query
+
+        select_fields: list
+            A list of fields to query over
+
+        use_json_encoder : bool
+            Whether to automatically convert documents to json encodable format
+
+        include_vector: bool
+            If True, includes vectors in the updating query
+
+        show_progress_bar: bool
+            If True, shows a progress bar
+
+        insert: bool
+            If True, inserts rather than updates an already-existing dataset
+
+        use_cache: bool
+        """
+        updating_args = {} if updating_args is None else updating_args
+        filters = [] if filters is None else filters
+        select_fields = [] if select_fields is None else select_fields
+
+        if not callable(update_function):
+            raise TypeError(
+                "Your update function needs to be a function! Please read the documentation if it is not."
+            )
+
+        if retrieve_chunk_size is None:
+            # Regardless of whether the cache is used, retrieve chunk size
+            # needs to be an integer.
+            retrieve_chunk_size = self._determine_optimal_chunk_size(
+                dataset_id, filters, select_fields
+            )
+            if not use_cache:
+                print(f"Setting retrieve_chunk_size to {retrieve_chunk_size}")
+
+        if (
+            retrieve_chunk_size is not None
+            and max(retrieve_chunk_size, 10_000) == retrieve_chunk_size
+            and retrieve_chunk_size != 10_000
+        ):
+            # Elasticsearch has an upperbound, 10,000, on the number of
+            # documents able to be pulled at once
+            print("Maximum value of retrieve_chunk_size is 10,000")
+            retrieve_chunk_size = 10_000
+
+        cache_path = (

```

Comment:
I think a cache path should exist inside a temporary file. this can be found in `appdirs` package 
`https://pypi.org/project/appdirs/` with the user_cache_dir